find_invalid_xml_chars.py

```
@@ -0,0 +1,115 @@
+
+
+def scan_string(path: str, value: str, findings: list):
+    bad = []
+    for idx, ch in enumerate(value):
+        cp = ord(ch)
+        if is_invalid_xml_char(cp):
+            bad.append((idx, cp))
+    if bad:
+        # Build a readable preview with offending chars shown as <0xNN>
+        preview_chars = []
+        for i, ch in enumerate(value[:120]):
```

Ruff (0.15.17)

[warning] Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)